pandas/indexes/multi.py

```diff
@@ -71,8 +71,10 @@ def __new__(cls, levels=None, labels=None, sortorder=None, names=None,
                 name=None, **kwargs):
 
         # compat with Index
-        if name is not None:
-            names = name
+        if name is None and hasattr(levels, 'name'):
+            name = levels.name
+        if isinstance(levels, MultiIndex):
+            return levels.copy(name=name, names=names, deep=copy)
```

Comment:
well did you test this? if this were to be supported then it should be the very first check.

I am not averse (and other Indexes do this), but needs testing.